src/transformers/generation/utils.py

```diff
@@ -1102,7 +1102,13 @@ def _validate_model_class(self):
     def _validate_model_kwargs(self, model_kwargs: Dict[str, Any]):
         """Validates model kwargs for generation. Generate argument typos will also be caught here."""
         # If a `Cache` instance is passed, checks whether the model is compatible with it
-        if isinstance(model_kwargs.get("past_key_values", None), Cache) and not self._supports_cache_class:
+        past = model_kwargs.get("past_key_values", None)
+        if isinstance(past, DynamicCache) and not self._supports_dynamic_cache_class:
+            raise ValueError(
+                f"{self.__class__.__name__} does not support an instance of `DynamicCache` as `past_key_values`. Please "
+                "check the model documentation for supported cache formats."
+            )
```

Comment:
I believe this new `if` and  the `_supports_dynamic_cache_class` model attribute are redundant: all models with `self._supports_cache_class = True` should support `DynamicCache`. As such, both this `if` and the new model attribute can be removed.

The exception is Jamba, which has a custom `Cache` (and no support for the legacy class). lmk if custom logic is needed for Jamba, so we can find a solution that doesn't require a new model attribute 🤗

Reply:
Ha yes, this has indeed become redundant since the clarification for cache supports attributes in https://github.com/huggingface/transformers/commit/9d889f870ed202de2859b0236895fe7f985a0c72! No need for this attribute anymore then 👌 I rebased my branch to apply latest changes in main, and remove the attribute